doc/introduction/templates.rst

```diff
@@ -105,6 +101,23 @@ The following embedding templates are available:
 
 .. _intro_ref_temp_params:
 
+Subroutines
+-----------
+
+Subroutines are simply a collection of (trainable) gates.
```

Comment:
The distinction of a subroutine might not be clear enough from this description. How is it different from a layer?